File: desafios/greeting-string.py
"""
    Created by Edwildson Coelho Rodrigues
    https://edwildson.dev
    https://github.com/edwildson
    https://www.linkedin.com/in/edwildson/

    Greeting - Os códigos nas demais linguagem podem ser encontrados em github.com:edwildson/visie-ps
"""
import re
import datetime

def remove_spaces(s):
    return s.strip()

def change_comma_to_exclamation(s):
    return s.replace(',', '!')

def greeting(s):
    hora = datetime.datetime.now().hour
    greet = ''

    if 0 <= hora < 12:
        greet = "Bom dia,"
    elif 12 <= hora < 18:
        greet = "Boa tarde,"
    else:
        greet = "Boa noite,"

    return re.sub(r'olá|Olá', greet, s)

def change_first_letter_to_a_in_string(string):
    if not isinstance(string, str):
        print('Entrada inválida.')
        return 0
    
    print('String antes: ', string)
    
    string = remove_spaces(string)
    string = change_comma_to_exclamation(string)
    string = greeting(string)
    
    print('result:', string)
    
    return string

change_first_letter_to_a_in_string(' Olá usuário, bem-vindo ao sistema  ')

File: desafios/greeting-string.js
/*
    Created by Edwildson Coelho Rodrigues
    https://edwildson.dev
    https://github.com/edwildson
    https://www.linkedin.com/in/edwildson/

    Greeting - Os códigos nas demais linguagem podem ser encontrados em github.com:edwildson/visie-ps
*/

function removeSpaces(string) {
    return string.trim();
}

function changeCommaToExclamation(string) {
    return string.replace(',', '!');
}

function greeting(string) {
    const hora = new Date().getHours();
    let greet = '';

    if (hora >= 0 && hora < 12) {
        greet = "Bom dia,";
    } else if (hora >= 12 && hora < 18) {
        greet = "Boa tarde,";
    } else {
        greet = "Boa noite,";
    }

    return string.replace(/olá|Olá/g, greet);
}

(function changeFirstLetterToAInArray(string) {
    if (!(typeof string === 'string' || string instanceof String)) {
        console.log('Entrada inválida.')
        return 0;
    }

    console.log('String antes: ', string)

    string = removeSpaces(string);
    string = changeCommaToExclamation(string);

    string = greeting(string);

    console.log('result:', string);

    return string;
})( ' Olá usuário, bem-vindo ao sistema  ')